Map/Map.cpp

```diff
@@ -0,0 +1,358 @@
+    //1- create two territories without connections
+    insertATerritory(dataA);
+    insertATerritory(dataB);
+    //2- make the connection between the two territories
+    connectTwoNodes(getV().end()[-1],getV().end()[-2]);
+}
+
+void Graph::connectTwoNodes(Node* A, Node* B){
+    string edgeName = A->getData().getTerritoryName() + B->getData().getTerritoryName(); //AB
+    //cout<<edgeName<<endl;
+    A->addEdge(edgeName);
+    B->addEdge(edgeName);
+}
+
+//-------------- Others --------------//
+bool Graph::areConnected(Node* A, Node* B){
+    string possibleEdge1 = A->getData().getTerritoryName() + B->getData().getTerritoryName(); //AB
+    string possibleEdge2 = B->getData().getTerritoryName() + A->getData().getTerritoryName(); //BA
+    
+    for(string edge : A->getE()){//loop through all of the edges of A
+        if(edge == possibleEdge1 || edge == possibleEdge2)
+            return true;
+         }
+
+    return false;
+}
+
+void Graph::validate(){
+    bool isErrorThrown = false;
+
+    //check that the map is a connected graph
+    try{
+        if(!DFS(getV(),getV()[0]))
+            throw std::logic_error("Unconnected Graph");
+
+    }catch(const std::exception& e){
+        isErrorThrown = true;
+        std::cerr<<e.what()<<endl;
+    }
+
+    //check if all continents are connected sub-graphs
+    try{
+        for(Continent* continent : getListOfContinents()){
+            vector<Node*> tempListOfTerritoriesInContinent;
+            //getting all territories with the same continent in a single vector
+            for(Node* node : getV()){
+                if(node->getData().getContinent() == continent)
+                    tempListOfTerritoriesInContinent.push_back(node);
+            }
+            
+            if(!DFS(tempListOfTerritoriesInContinent,tempListOfTerritoriesInContinent[0]))
+                throw std::logic_error("Unconnected Sub-Graph: "+ continent->getContinentName() + " continent is not connected.");
+
+            tempListOfTerritoriesInContinent.clear();//earse the vector contentss
+        }
+
+    }catch(const std::exception& e){
+        isErrorThrown = true;
+        std::cerr<<e.what()<<endl;
+    }
+
+    //check if each country belongs to one and only one continent
+    try{
+        for(Node* node : getV()){
+            int countNumOfMatches = 0; //count how many continents associate with each territory
+            for(Continent* Continent : getListOfContinents()){
+                if(node->getData().getContinent() == Continent)
+                    countNumOfMatches++;
+            }
+
+            if(countNumOfMatches != 1)
+                throw std::logic_error("No 1-1 associate between "+ node->getData().getTerritoryName() + " and the continents.");
+        }
+    }catch(const std::exception& e){
+        isErrorThrown = true;
+        std::cerr<<e.what()<<endl;
+    }
+
+    //will terminate the program if an error was thrown
+    if(isErrorThrown)
+        exit(0);
```

Comment:
Maybe use EXIT_FAILURE instead of 0 to indicate that something unexpected/wrong happened. It isn’t a normal exit.
exit(EXIT_FAILURE);